web/yo/app/scripts/services/databaseconnector.js

```diff
@@ -719,35 +721,35 @@ angular.module('oncokbApp')
                 return deferred.promise;
             }
 
-            function updateHistory(historyData) {
-                if (!$rootScope.model.getRoot().get('history')) {
-                    $rootScope.model.getRoot().set('history', $rootScope.model.createList());
-                    return;
-                }
-                var apiHistory = $rootScope.model.getRoot().get('history').get('api');
-                if (!apiHistory || !_.isArray(Array.from(apiHistory))) {
-                    apiHistory = [];
-                } else {
-                    apiHistory = Array.from(apiHistory);
-                }
-                if (apiHistory.length > 3000) {
-                    // send email to the oncokb dev account with the oldest 500 records
-                    var historyToRemove = apiHistory.splice(0, 500);
-                    sendEmail({sendTo: 'dev.oncokb@gmail.com', subject: 'OncoKB Review History', content: JSON.stringify(historyToRemove)},
-                        function(result) {
-                            console.log('sent old history to oncokb dev account');
-                        },
-                        function(error) {
-                            console.log('fail to send old history to oncokb dev account', error);
-                        }
-                    );
-                }
-                apiHistory.push({
-                    admin: user.name,
-                    timeStamp: new Date().getTime(),
-                    records: historyData
+            function updateHistory(hugoSymbol, historyData) {
```

Comment:
Passing hugoSymbol down to here is not a good way to share data. Please assign current gene hugoSymbol to a rootScope variable, just like the way how $rootScope.model is assigned